third_party/move/move-core/types/src/resolver.rs

```diff
@@ -33,29 +36,43 @@ pub trait ModuleResolver {
 ///                       are always structurally valid)
 ///                    - storage encounters internal error
 pub trait ResourceResolver {
-    fn get_resource(
+    fn get_resource_with_metadata(
         &self,
         address: &AccountAddress,
         typ: &StructTag,
+        metadata: &[Metadata],
     ) -> Result<Option<Vec<u8>>, Error>;
 }
 
 /// A persistent storage implementation that can resolve both resources and modules
-pub trait MoveResolver: ModuleResolver + ResourceResolver {}
+pub trait MoveResolver: ModuleResolver + ResourceResolver {
+    fn get_resource(
+        &self,
+        address: &AccountAddress,
+        typ: &StructTag,
+    ) -> Result<Option<Vec<u8>>, Error> {
+        self.get_resource_with_metadata(address, typ, &self.get_module_metadata(&typ.module_id()))
+    }
```

Comment:
I think this should be moved into `ResourceResolver` so that we keep the old interface intact.

Reply:
but module_metadata lives in moduleResolver, the fact that we split it is not ideal. We could make get_module_metadata part of ResourceResolver but that seems wrong as well

Reply:
My bad. Forgot that you need to access get_module_metadata.